clients/src/test/java/org/apache/kafka/test/MockSelector.java

```diff
@@ -72,6 +77,24 @@ public void close(String id) {
         }
     }
 
+    /**
+     * Simulate a server disconnect. This id will be present in {@link #disconnected()} on
+     * the next {@link #poll(long)}.
+     */
+    public void serverDisconnect(String id) {
+        this.disconnected.put(id, ChannelState.LOCAL_CLOSE);
```

Comment:
This is ok in terms of the test, but `LOCAL_CLOSE` is the state used for client-side disconnection. For server-side disconnections, the state would be whatever the client state was at the time the disconnection was detected. Could just use `ChannelState.READY`.